src/algebra/opposites.lean

```diff
@@ -364,3 +364,31 @@ rfl
 lemma units.coe_op_equiv_symm {R} [monoid R] (u : (units R)ᵒᵖ) :
   (units.op_equiv.symm u : Rᵒᵖ) = op (u.unop : R) :=
 rfl
+
+/-- A hom `α →* β` can equivalently be viewed as a hom `αᵒᵖ →* βᵒᵖ`. This is the action of the
+(fully faithful) `ᵒᵖ`-functor on morphisms. -/
+@[simps]
+def monoid_hom.op {α β} [mul_one_class α] [mul_one_class β] :
+  (α →* β) ≃ (αᵒᵖ →* βᵒᵖ) :=
+{ to_fun    := λ f, { to_fun := op ∘ f ∘ unop, map_one' := by simp, map_mul' := λ x y, by simp },
+  inv_fun   := λ f, { to_fun := unop ∘ f ∘ op, map_one' := by simp, map_mul' := λ x y, by simp },
+  left_inv  := λ f, by { ext, simp },
+  right_inv := λ f, by { ext, simp } }
```

Comment:
The term mode proof is longer, but elaborates 20x faster for me:
```suggestion
{ to_fun    := λ f, { to_fun := op ∘ f ∘ unop,
                      map_one' := unop_injective f.map_one,
                      map_mul' := λ x y, unop_injective (f.map_mul _ _) },
  inv_fun   := λ f, { to_fun := unop ∘ f ∘ op,
                      map_one' := op_injective f.map_one,
                      map_mul' := λ x y, congr_arg unop (f.map_mul (op _) (op _)) },
  left_inv  := λ f, by { ext, refl },
  right_inv := λ f, by { ext, refl } }
```

Reply:
Nice!